### PR TITLE
PCVL-657 Send NoiseModel & PostSelect in payload

### PR DESCRIPTION
**Pull request by ericbrts**

### CODE REVIEW

File: perceval/serialization/deserialize.py [hunk 1]
```diff
@@ -145,38 +158,22 @@ def deserialize(obj):
...
-            r = deserialize_circuit(obj)
-        elif cl == "NoiseModel":
-            r = deserialize_noise_model(sobj)
-        else:
-            raise NotImplementedError(f"No deserializer found for {cl}")
+            obj = b64decode(obj[len(ZIP_PREFIX):])
+            obj = decompress(obj).decode('utf-8')
+
+        p = obj[len(PCVL_PREFIX):].find(SEP)
+        class_obj = obj[6:p+6]
```
Comment (raksharuia) [inline perceval/serialization/deserialize.py:169]: I would suggest replacing the number '6' in the code, perhaps using `len(PCVL_PREFIX)` to allow for a quicker understanding.

File: tests/test_job.py [hunk 2]
```diff
@@ -134,92 +130,10 @@ def test_get_res_run_async():
...
-            "status_message": None
-        }
-
-    def get_job_results(self, job_id: str):
-        time.sleep(self._SLEEP_SEC)
-        return {'results': json.dumps({
-            'results': serialize(_REMOTE_JOB_RESULTS),
-            'physical_perf': 1
-        })}
+from _mock_rpc_handler import MockRPCHandler, REMOTE_JOB_DURATION, REMOTE_JOB_RESULTS, REMOTE_JOB_CREATION_TIMESTAMP, \
```
Comment (raksharuia) [inline tests/test_job.py:135]: Wondering -
Should we split this file into 2 test files? `test_job.py` and `test_remote_job.py`? I kept trying to find import statements at the top hahaha :-D :-(

**Review by MarionQuandela — CHANGES_REQUESTED** [review]
The worker accept the modifications without complaining (and do nothing) don't we want to add some key verification worker side ?
Some typo:
- connectible -> connectable
- distinguishible -> distinguishable

File: perceval/components/source.py [hunk 3]
```diff
@@ -76,6 +76,8 @@ def __init__(self,
 
     @staticmethod
     def from_noise_model(noise: NoiseModel):
+        if noise is None:
+            return Source()
```
[the remark this answers is not in the document]
Comment (ericbrts) [reply perceval/components/source.py:80]: Let's say `Source` previously was our noise data class and computation class at the same time.
Now, it's split into two classes, but of course `NoiseModel` and `Source` are going to share some parameters, as they both handle noise parameters.

File: perceval/components/processor.py [hunk 4]
```diff
@@ -93,6 +93,30 @@ def noise(self, nm):
...
+        return self._source
+
+    @source.setter
+    def source(self, source: Source):
+        r"""
+        :param source: A Source instance to use as the new source for this processor.
+        Input distribution is reset when a source is set, so `with_input` has to be called again afterwards.
+        """
+        self._source = source
+        self._inputs_map = None
```
[the remark this answers is not in the document]
Comment (ericbrts) [reply perceval/components/processor.py:118]: Historical naming - even though it's a private member, I know it's inspected from the outside sometimes, so I wouldn't like to change it.
The naming refers to:
* map: our distributions being dictionaries (or maps),
* inputs: the plural of input (it's probably improper English) to express the fact we now have several inputs to handle (that's how noisy inputs are represented).

**Pull request merged**